### ISSUE TITLE
Borrow some Livestreamer smarts

### ISSUE THREAD

**Issue by alanmcramer** [Fleker/CumulusTV#112]
Would be cool to be able to provide a HTTP URL and have Cumulus TV use a Livestreamer plugin to be able to decode a URL directly. Some web providers use a wmsAuthSign which is generated on the fly and it would be great to have something to decode and generate these on the fly.

**Comment by Fleker** [comment]
I agree this would be a good feature and it's something I tried to support
through the plugin model. Nobody has taken up the job at this point.

**Comment by alanmcramer** [comment]
LiveStreamer is written in Python and so are the plugins that use it.  Would it be hard to have something that could parse the Livestreamer plugins and use them within Cumulus TV?

**Comment by Fleker** [comment]
It shouldn't be too difficult, although the Python may need to be ported to
Java.

**Comment by MrJinx123** [comment]
Is there a tool you could recommend to convert IPTV m3u files to a JSON format compatible with your sofware? It's rather time consuming to manually enter and fix all the channel #'s. Especially as often as the links change.

**Comment by MrJinx123** [comment]
Fleker, Have you had a chance to look at this GitHub Project? https://github.com/notz/iptv-live-channels . I was able to build it and repoint it to my own web server to serve up IPTV channels, It's not complete.. It only supports M3U8 types. Doesn't support browser strings (UAE) needed to get sources to offer higher quality streams. It's also lacking stalker support Mac address and donation support. The playlist format is similar to M3U but has a few extra fields in it. I would like to get away from KODI or the STB emulator and use Live Channels for all of it even if it means buying an app....  
I know you are a big fan of live channels as well. Seen your articles on Android.tv

**Comment by Fleker** [comment]
You can import m3u8 files into the app. I don't know of a valid tool for converting IPTV to this JSON format.

I have looked at the project you posted. I've adapted some of the code they've used and have been looking at their parsers as well.
